src/frontend/src/handler/create_source.rs

```diff
@@ -111,22 +117,37 @@ pub async fn handle_create_source(
     let (mut columns, pk_column_ids, row_id_index) =
         bind_sql_table_constraints(column_descs, pk_column_id_from_columns, stmt.constraints)?;
 
-    let mut with_properties = context.with_options.inner().clone();
-
+    let with_properties = context.with_options.inner().clone();
+    const UPSTREAM_SOURCE_KEY: &str = "connector";
+    // confluent schema registry must be used with kafka
+    let is_kafka = with_properties
+        .get("connector")
+        .unwrap_or(&"".to_string())
+        .to_lowercase()
+        .eq("kafka");
+    if !is_kafka
+        && matches!(
+            &stmt.source_schema,
+            SourceSchema::Protobuf(ProtobufSchema {
+                use_schema_registry: true,
+                ..
```

Comment:
```suggestion
                ..
```

Looks like we don't need this?

Reply:
Do you mean that avro/protobuf can only be used with kafka? But actually there may be other data sources (e.g. other message queues). I mean schema registry can only be used with kafka.